astarte/device/device_grpc.py

```diff
@@ -0,0 +1,579 @@
+# This file is part of Astarte.
+#
+# Copyright 2023 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import collections.abc
+import json
+import logging
+import queue
+from collections import namedtuple
+from datetime import datetime, timezone
+from enum import Enum
+from threading import Thread
+
+# pylint: disable=no-name-in-module
+import grpc
+from astarteplatform.msghub.astarte_message_pb2 import AstarteMessage, AstarteUnset
+from astarteplatform.msghub.astarte_type_pb2 import (
+    AstarteBinaryBlobArray,
+    AstarteBooleanArray,
+    AstarteDataType,
+    AstarteDataTypeIndividual,
+    AstarteDataTypeObject,
+    AstarteDateTimeArray,
+    AstarteDoubleArray,
+    AstarteIntegerArray,
+    AstarteLongIntegerArray,
+    AstarteStringArray,
+)
+from astarteplatform.msghub.message_hub_service_pb2_grpc import MessageHubStub
+from astarteplatform.msghub.node_pb2 import Node
+from google.protobuf.timestamp_pb2 import Timestamp
+from grpc import ChannelConnectivity
+from grpc._channel import _MultiThreadedRendezvous
+
+# pylint: enable=no-name-in-module
+from astarte.device.device import Device
+from astarte.device.exceptions import (
+    DeviceConnectingError,
+    DeviceDisconnectedError,
+    ValidationError,
+)
+from astarte.device.interface import Interface
+from astarte.device.mapping import Mapping
+
+
+class ConnectionState(Enum):
+    """
+    Possible connection states for a device.
+    """
+
+    CONNECTING = 1
+    CONNECTED = 2
+    DISCONNECTED = 3
+
```

Comment:
The ConnecctionState enum and related methods is defined also in the `device-mqtt.py`, What do you think to move in the Abstract Class? Is this possible?